docs/articles/graphql.md

```diff
@@ -0,0 +1,739 @@
+# Using GraphQL for Enhanced Searches
+
+> :point_right: A GraphQL backend server within zot's registry search engine provides efficient and enhanced search capabilities. You can submit a GraphQL structured query as an API call or you can use a browser to access the GraphQL Playground, an interactive graphical environment for GraphQL queries.  
+
+<a name="howto"></a>
+## How to use GraphQL for search queries
+
+GraphQL is a query language for APIs. A GraphQL server, as implemented in zot's registry search engine, executes GraphQL queries that match schema recognized by the server. In response, the server returns a structure containing the requested information. The schema currently recognized by zot are those that correspond to the queries listed in [What GraphQL queries are supported](#queries).
+
+> :bulb: To learn more about GraphQL, see these resources:
+>
+>  - [Introduction to GraphQL](https://graphql.org/learn/)
+>
+>  - [The Fullstack Tutorial for GraphQL](https://www.howtographql.com/)
+
+To perform a search, compose a GraphQL structured query for a specific search and deliver it to zot using one of the methods described in the following sections.
+
+For examples of GraphQL queries supported in zot, see [Examples of zot searches using GraphQL](#examples).
+
+### Using the search API directly
+
+You can submit a GraphQL structured query as the HTML data payload in a direct API call using a shell tool such as cURL or Postman. GraphQL queries are sent to the zot `search` extension API:
+
+    /v2/_zot/ext/search
+
+The following example submits a zot GraphQL query using cURL:
+
+```shell
+curl -X POST -H "Content-Type: application/json" --data '{ "query": "{ ImageListForCVE (id:\"CVE-2002-1119\") { Results { Name Tags } } }" }' http://localhost:8080/v2/_zot/ext/search
+```
+
+The reply to your query is returned as a JSON payload in the HTML response.
+
+### Using the GraphQL Playground
+
+> :pencil2: The GraphQL Playground feature is available only in a `binary-debug` zot build or when the zot registry was built with the `debug` extension label.
+
+The GraphQL Playground is an interactive graphical web interface for GraphQL hosted by the zot registry server. 
+
+The GraphQL Playground is reachable by a browser at the following zot API:
+
+    /v2/_zot/debug/graphql-playground#
+
+For example, if your zot server is located at http<nolink>://localhost:8080, the GraphQL Playground can be accessed by your browser at this URL:
```

Comment:
```suggestion
For example, if your zot server is located at http://localhost:8080, the GraphQL Playground can be accessed by your browser at this URL:
```

Reply:
The nolink tag is there so that the URL renders as text and not a hyperlink.  If you think it should be an active link, I'll remove the tag.

Reply:
Nevermind then.